parseFhir.py

```diff
@@ -56,115 +54,308 @@ def getJsonValue(lnjsn,ln):
+            }
+        )
+    else:
+        for res in resources:
+            resource_type = json.loads(res)["resourceType"]
+            if resource_type not in found_resources:
+                # find which configPath is the right one
+                found_resources.append(resource_type)
+
+        for i, path in enumerate(config_paths):
+            parsed_paths.append(found_resources[i])
+            config = configparser.ConfigParser()
+            config.read(path)
+
+            write_mode, anchor, header, paths, leng = get_config_options(config)
+
+            thisRow = [""] * leng
+
+            if config.has_option("GenConfig", "resourceType"):
+                matched_fhir_resources = [
+                    res
+                    for res in resources
+                    if json.loads(res)["resourceType"] == found_resources[i]
+                ]
+                if len(matched_fhir_resources) == 0:
+                    print(
+                        f"Warning: no resources found for {found_resources[i]} resource"
+                    )
+                    continue
+                outputPath = f"{found_resources[i]}.csv"
+                csv_file = open(outputPath, write_mode, newline="")
+                csv_writer = csv.writer(
+                    csv_file,
+                    delimiter=",",
+                    escapechar="\\",
+                    quoting=csv.QUOTE_ALL,
+                )
+
+                if write_mode == "w":
+                    csv_writer.writerow(header)
+                    write_mode = "a"
+
+                for matched_resource in matched_fhir_resources:
+                    output.append(
+                        {
+                            "resource": matched_resource,
+                            "inputPath": inputPath,
+                            "thisRow": thisRow,
+                            "leng": leng,
+                            "csv_writer": csv_writer,
+                            "paths": paths,
+                            "anchor": anchor,
+                        }
+                    )
+            else:
+                print(f"Error: resourceType key for {path} missing!")
+                exit()
+
+        # check to see if any resources were missed
+        missed_resources = []
+        for res in get_fhir_resources_types_from_bundle(inputPath):
+            if not filter_resources(res, parsed_paths):
+                missed_resources.append(res)
+
+        if len(missed_resources) > 0:
+            m_out = ", ".join(map(str, missed_resources))
+            print(
+                f"Warning: the following resources were found but not parsed: {m_out}"
+            )
+
+    return output
+
+
+def parse_resource(
+    input_resource, input_path, row, config_length, csv_writer, paths, anchor
+):
+
+    newInput = []
+    if ".json" in input_path:
+        newInput = convert_json_resource_to_ndjson(input_resource)
+    else:
+        newInput = input_resource
```

Comment:
I think this is fine but this works by checking the filextension of the input - so if you need it .json that is .ndjson it will break right?

Reply:
Correct, right now if you have a `.ndjson` file with `.json` in it it will break